src/components/shortestpathfinder/UserCase2.java

```diff
@@ -0,0 +1,22 @@
+package components.shortestpathfinder;
+
+public final class UserCase2 {
+
+    public static void main(String[] args) {
```

Comment:
This seems really similar to the previous one. I'm not sure of a good way to use the component another way, but maybe you could make two graphs and compare them (like maybe if there was an extra location in the graph to show how their might be a new shorter path).